packages/data/addon/mirage/routes/bard-lite.js

```diff
@@ -227,6 +232,99 @@ export default function(
 
     let missingIntervals = request.queryParams.metrics.includes('uniqueIdentifier') ? MISSING_INTERVALS : undefined;
 
+    //lets do subtotal total stuff
+    const processedDims = [];
+    if (request.queryParams.rollupTo || request.queryParams.rollupGrandTotal) {
+      let rollupDimensions = [];
+      if (request.queryParams.rollupTo) {
+        rollupDimensions = request.queryParams.rollupTo.split(',');
+      }
+      let groupedRows = [[...rows]];
+      const metrics = parseMetrics(request.queryParams.metrics);
+      const getMask = rolledupDims => {
+        const rollupDimensionMask = ['dateTime', ...dimensions];
+        const flagMap = rollupDimensionMask.map(rolledDim => (rolledupDims.includes(rolledDim) ? 1 : 0));
+        return parseInt(flagMap.join(''), 2);
+      };
+      let grandTotalRow;
+
+      if (request.queryParams.rollupGrandTotal) {
+        grandTotalRow = rows.reduce((acc, row) => {
+          if (acc === null) {
+            acc = {};
+            Object.keys(row).forEach(key => (acc[key] = null));
+            if (hasRollUpDim) {
+              acc.__rollupMask = 0;
+            }
+          }
+
+          metrics.forEach(metric => {
+            acc[metric] = acc[metric] + Number(row[metric]);
+          });
+
+          if (hasRollUpDim) {
+            row.__rollupMask = getMask(['dateTime', ...dimensions]);
```

Comment:
can `getMask(['dateTime', ...dimensions])` be a const? could help with readability